### PR TITLE
instrumentation/otelgrpc: rename Typ to Type

### PR DESCRIPTION
**Pull request by zchee**
`Typ` is not a reserved name. Use the correct name instead of.

### CODE REVIEW

**Comment by zchee** [comment]
cc: @ymotongpoo

**Comment by dmathieu** [comment]
Can you fix the lint issue? Also, this is changing a public attribute. So it will require a changelog entry.

**Comment by zchee** [comment]
will do, thanks

**Comment by zchee** [comment]
@dmathieu PTAL.

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ### Changed
 
+- Rename Typ to Type on `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc#InterceptorInfo` field. (#2688)
```
Comment (MrAlias) [inline CHANGELOG.md:13]:
```suggestion
- Rename the `Typ` field of `"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc".InterceptorInfo` to `Type`. (#2688)
```

**Comment by zchee** [comment]
@MrAlias PTALA.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**